### PR TITLE
KRED-749 ->  Implement BRI, BNI, CIMB, Permata charge and Mandiri Bill Charge

### PR DESCRIPTION
**Pull request by rikyhidayat21**

### CODE REVIEW

File: lib/midtrans_api/api/transaction/charge_echannel.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module MidtransApi
+  module Api
+    module Transaction
+      class ChargeEchannel < MidtransApi::Api::Base
+        PATH          = 'charge'
+        PAYMENT_TYPE  = 'echannel'
```
Comment (VincentWijaya) [inline lib/midtrans_api/api/transaction/charge_echannel.rb:8]: is echannel type for mandiri ki? Please add comment for it if echannel for mandiri

File: lib/midtrans_api/api/transaction/charge.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module MidtransApi
```
Comment (VincentWijaya) [inline lib/midtrans_api/api/transaction/charge.rb:3]: please add comment of VA list that `bank_transfer` type supported

File: lib/midtrans_api/api/transaction/charge_echannel.rb (hunk 1 above)
Comment (rikyhidayat21) [reply lib/midtrans_api/api/transaction/charge_echannel.rb:8]: I already update the details at README.md file ya cent, please help to check

**Review by VincentWijaya — APPROVED** [review]

File: lib/midtrans_api/api/transaction/charge_echannel.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module MidtransApi
+  module Api
+    module Transaction
+      class ChargeEchannel < MidtransApi::Api::Base
```
[the remark this answers is not in the document]
Comment (rikyhidayat21) [reply lib/midtrans_api/api/transaction/charge_echannel.rb:6]: no, the response is also different mas. That's why I differentiate the class.

[the remark this answers is not in the document]
Comment (okfriansyah-moh) [reply lib/midtrans_api/api/transaction/charge_echannel.rb:6]: The different in response is in this one right?
  "va_numbers": [
    {
      "bank": "bni",
      "va_number": "8578000000111111"
    }
  ]
  
  and for echannel is 
    "bill_key": "990000000260",
  "biller_code": "70012",
  
  Can we make it into 1 class and make differentiation based on that request and response specific ?

[the remark this answers is not in the document]
Comment (rikyhidayat21) [reply lib/midtrans_api/api/transaction/charge_echannel.rb:6]: let's see if it's possible, I'll try update it

[the remark this answers is not in the document]
Comment (rikyhidayat21) [reply lib/midtrans_api/api/transaction/charge_echannel.rb:6]: updated, please help to check ya @okfriansyah-moh

**Review by okfriansyah-moh — APPROVED** [review]

**Pull request merged**